src/ui/pages/SwapForm/SwapForm.tsx

```diff
@@ -501,6 +519,9 @@ export function SwapForm() {
         height={innerHeight >= 750 ? '70vh' : '90vh'}
         containerStyle={{ display: 'flex', flexDirection: 'column' }}
         renderWhenOpen={() => {
+          if (!currentTransaction || approveTransactionQuery.isLoading) {
```

Comment:
Looks like we should attempt to render `TransactionConfirmationView` if there is no `currentTransaction`.
The issue here is that `currentTransaction` won't be available until `approveTransactioinQuery` is loaded

Reply:
Setting `keepPreviousData: true` kind of helps but there is still an issue with resetting allowance